query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -173,4 +228,15 @@ public ResultSetMetadata getResultSetMetadata() {
   public LinkedHashSet<Expression> getAllSelections() {
     return this.allSelections;
   }
+
+  public Optional<Double> getTimeSeriesPeriod() {
+    if (this.timeSeriesPeriod.isZero()) {
+      return Optional.empty();
+    }
+    return Optional.of((double) this.timeSeriesPeriod.getSeconds());
```

Comment:
We're so close! just 
```suggestion
    return Optional.of(this.timeSeriesPeriod);
```